### DISCUSSION TITLE
Need any changes for dumping gba cartridges?

### DISCUSSION THREAD

**Discussion by joaopauloap** [wodowiesel/GB-Dumper#4] [Q&A] [answered]
Hi. I've build your project for dumping some gba saves, but i can't make it work, only shows "not found" in mbc, rom and ram size. in addition to 3.3v logic, do i need make any modifications on wiring? i think gba carts made use of CS2, but i've not seen this pin being used on code.

Beside that, i was developing a gba save dumper before seeing your project. but mine its very simple:
https://github.com/joaopauloap/arduino-gba-dumper

i want figure why this works with only one cartridge, a 369 in 1 (i'm testing with Mario advanced 2) and not with my others cartridges, like pokemon fire red (at least to 0x1FFF) or zelda minish cap (which is 8kb too). i think could be the memory type. can you help me?

this was my ideia to read 8kb of ram:

 digitalWrite(CS2, LOW);
  for (unsigned long addr = 0; addr < 0x1FFF; addr++) {
    latchAddress(addr);
    digitalWrite(RD, LOW);
    delayMicroseconds(2);
    digitalWrite(RD, HIGH);
    int b = readDataBus();
    Serial.write(b);
  }

**Comment by wodowiesel** [comment]
@joaopauloap  yes, you do need changes. im working atm on the solution in the schematics + pcb, 
they are not up to date with the newer codeincluding jumpers and switches for cs2 and irq and reset.
i  would update it this week, so still in WIP.
theoretically cs2 is not needed according to some sources but for full compatibility i have it already in.
the memory typeis read by CS  number 1. thats what i could get. not actually sure what data is being transfered on CS2.
a debug print out would be nice if you could provide it.
thx for the code suggestion i give it a try on my side! for now the code is with gb dmg compatible. 

i think some pokemon are special ,there are some games that might need extra adaptions.
would you please try to dump a normal classic gb &  normal gba rom + saves? 

> Hi. I've build your project for dumping some gba saves, but i can't make it work, only shows "not found" in mbc, rom and ram size. in addition to 3.3v logic, do i need make any modifications on wiring? i think gba carts made use of CS2, but i've not seen this pin being used on code.
> 
> Beside that, i was developing a gba save dumper before seeing your project. but mine its very simple: https://github.com/joaopauloap/arduino-gba-dumper
> 
> i want figure why this works with only one cartridge, a 369 in 1 (i'm testing with Mario advanced 2) and not with my others cartridges, like pokemon fire red (at least to 0x1FFF) or zelda minish cap (which is 8kb too). i think could be the memory type. can you help me?
> 
> this was my ideia to read 8kb of ram:
> 
> digitalWrite(CS2, LOW); for (unsigned long addr = 0; addr < 0x1FFF; addr++) { latchAddress(addr); digitalWrite(RD, LOW); delayMicroseconds(2); digitalWrite(RD, HIGH); int b = readDataBus(); Serial.write(b); }

**Comment by wodowiesel** [comment]
hey @joaopauloap,

i uploaded/released the hardware and software changes  and documentations.
hope it helps you and may need minor cs2, irq function debugging. hardware-wise it should be fine.
you can give it to a pcb manufacturer and test it,  i will order some too

do you have some new informations about  the dumping process via c2 or achieved some  dumps meanwhile?
would be cool to hear some feedback :) always appreciated!

**Reply by joaopauloap** [reply, marked as answer]
> hey @joaopauloap,
> 
> i uploaded/released the hardware and software changes and documentations. hope it helps you and may need minor cs2, irq function debugging. hardware-wise it should be fine. you can give it to a pcb manufacturer and test it, i will order some too
> 
> do you have some new informations about the dumping process via c2 or achieved some dumps meanwhile? would be cool to hear some feedback :) always appreciated!

Hello,  sorry for late.
i haven't tried again dumping gba cartridges with your project, but i promise i will. and i just only have gba cartridges. i think that i've discovered something about my previous question about my own dumper project where i'd asked why can i get dump save from a "369 in 1" cartridge but not of the standalones. 
On the "369 in 1" i got the save of any game that has it there, like Mario Advance, Pokemon Fire Red, etc. However, I tried with the standalone games I own (Pokemon Fire Red, Zelda Minish Cap, Metroid Zero Mission), but without success.

Maybe the answer could be that in these i need send some read operation commands to the flash chip. look the final page of this pdf:
https://github.com/jaro10008/GBA_cartridge_dumper/blob/master/Signal%20diagrams.pdf

it also says about the use of cs2 pin on the first page.

**Reply by wodowiesel** [reply]
thx for reply and no hurry^^
would be intresting to see newer code if you can provide ?
think it needs a mix of cs1 nd cs2, cause normal gb games only have cs1 and works with most mbc types. 
it just might be a software problem, not a physical anymore but i check the schematic from the link
and  the pdf is quite useful including states, pins edges etc.! ill try it myself this or next weekend, i have lot of work.

until then have a nice day
